libcontainer/standard_init_linux.go

```diff
@@ -48,6 +48,10 @@ func (l *linuxStandardInit) Init() error {
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 	if !l.config.Config.NoNewKeyring {
+		if err := label.SetKeyLabel(l.config.ProcessLabel); err != nil {
```

Comment:
Just to double-check -- are you sure this needs to be done *before* we create a new session? (Is `SetKeyLabel` setting what the label will be for all future keys or the label for the current key?)

Reply:
Yes this says to the kernel, any new keyrings that get created by this process and its children should get created with the specified label.
When the new keyring gets created it gets created with the correct label.
Later on the defaulf SetKeyLabel("") tells the kernel to go back to the default labeling.

Reply:
BTW I have been testing this patch with the updated kernel, which allows sepate kernel keyrings for each new UserNamespace.

Reply:
I am not clear why we need to reset it back. Shouldn't it pass down to the container process?

Reply:
Yes it will, the reset is on defer.  Not really important for runc, but if people vendor in this code, then we would want to make sure other code paths don't accidently set the label to this container.

Reply:
Yup, we've discussed the `defer` handling for SELinux in the past. Even though it doesn't really make sense for `runc` it avoids people vendoring it and shooting themselves in the foot (not that SELinux is the only thing that is a massive foot-gun in `libcontainer`).